### PR TITLE
Develop automatic retry of getting a open port

### PR DESCRIPTION
**Pull request by Juli199696**
This helps to retry to get a open port for a server as it will ask the network again for the port to be open.

### CODE REVIEW

**Comment by Juli199696** [comment]
Please someone do it with a while command as i dont get it to work with while and i think goto is a bad command for this.

File: src/network/network.cpp [hunk 1]
```diff
@@ -285,17 +287,20 @@ bool Network::BeginServer(unsigned short port, const char* address)
...
 	if (setsockopt(listening_socket, IPPROTO_IPV6, IPV6_V6ONLY, (const char*)&value, sizeof(value)) != 0) {
 		log_error("IPV6_V6ONLY failed. %d", LAST_SOCKET_ERROR());
 	}
+	
+	if (bind(listening_socket, (sockaddr *)&ss, ss_len) != 0)
+	{
+		//closesocket(listening_socket);
+		log_error("Unable to bind to socket. Retrying");
+		Sleep(5000);
+		goto retry;
```
Comment (LinusU) [inline src/network/network.cpp]: *maybe not technically true, still*

**Comment by IntelOrca** [comment]
This isn't an appropriate thing to do.

* It will get stuck in an infinite loop if the port is never freed. Lets say I choose port 80 when I already have a web server running.
* Its not asynchronous, the game will just hang until it successfully binds.
* It isn't really the game's problem if the port is bindable, so why should it retry?
* I have also unfortunately just completely refactored the socket code, so this would have to be changed.

**Comment by Juli199696** [comment]
Well okay, so i need to life with that thing. 
1. It also stay open if there is no port to catch
2. Sure it should hang there and wait for the port to be open
3. Maybe its possible to make sure that on crash or ending the process of the game/server the ports get the propper socket closed command. 
4. Good job with the refactoring!

**Comment by Goddesen** [comment]
I agree with @IntelOrca on not having the game automatically retry for sockets. If a server admin wants to have the server automatically restart on crash that can easily be scripted and manipulated in any way the admin sees fit.

However, I do think it is wrong for a program in both headless and host mode to enter the game loop when it fails to bind to a socket - at least as long as #3774 or something similar remains unimplemented.

**Comment by Juli199696** [comment]
Than please at least make the game close the Socket properly on exit or Crash. Maybe a cin command for the headless mode would be great with a exit or stop command to stop the server correctly instead of sigterm or just pressing the x on Windows as i still got the problem with the used socket after close or crash the server

**Comment by fromthere** [comment]
@Juli199696 you should open new issue for that.

**Comment by IntelOrca** [comment]
@Juli199696 how can it close the socket on crash? Its crashed... it can't do anything.

**Pull request closed without merging**